### PR TITLE
Folders API Add support for max_results and next_cursor

### PR DESCRIPTION
**Pull request by RTLcoil**

### CODE REVIEW

File: Shared.IntegrationTests/AdminApi/ManageFoldersTest.cs [hunk 1]
```diff
@@ -2,70 +2,249 @@
...
+
+            AssertDeleteFolderError(deletionRes);
+
+            m_cloudinary.DeleteResourcesByPrefix(_subFolder1);
+
+            deletionRes = m_cloudinary.DeleteFolder(_subFolder1);
+
+            AssertDeleteFolder(deletionRes);
+
+            DeleteFoldersForTests();
```
Comment (const-cloudinary) [inline Shared.IntegrationTests/AdminApi/ManageFoldersTest.cs:46]: Please move this and all `DeleteFoldersForTests()` to `TearDown()`.

Here it is never executed if at least one assertion fails and folders are not deleted.

Comment (RTLcoil) [reply Shared.IntegrationTests/AdminApi/ManageFoldersTest.cs]: Updated

File: Shared/Cloudinary.cs [hunk 2]
```diff
@@ -1081,38 +1081,45 @@ private string GetUploadUrl(string resourceType)
...
 
         /// <summary>
         /// Gets a list of subfolders in a specified folder asynchronously.
         /// </summary>
         /// <param name="folder">The folder name.</param>
         /// <param name="cancellationToken">(Optional) Cancellation token.</param>
+        /// <param name="parameters">(Optional) Parameters for managing folders list.</param>
         /// <returns>Parsed result of folders listing.</returns>
-        public Task<GetFoldersResult> SubFoldersAsync(string folder, CancellationToken? cancellationToken = null)
+        public Task<GetFoldersResult> SubFoldersAsync(string folder, CancellationToken? cancellationToken = null, GetFoldersParams parameters = null)
```
Comment (const-cloudinary) [inline Shared/Cloudinary.cs:1116]: Parameters order is inconsistent with other methods `(subject, parameters, cancellation token)`
Please fix the order and add and overloaded method ` public Task<GetFoldersResult> SubFoldersAsync(string folder, CancellationToken? cancellationToken = null)` for backwards compatibility.

File: Shared/ApiShared.cs [hunk 3]
```diff
@@ -13,7 +13,7 @@
     using System.Threading.Tasks;
     using CloudinaryDotNet.Actions;
     using Newtonsoft.Json;
-
+
```
Comment (const-cloudinary) [inline Shared/ApiShared.cs]: Please make sure to use linux line ending(LF) and not windows(CRLF) and please do not mix them.

**Pull request merged**